src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1120,4 +1123,79 @@ ValidMPTIssuance::finalize(
         mptokensCreated_ == 0 && mptokensDeleted_ == 0;
 }
 
+//------------------------------------------------------------------------------
+
+void
+ValidPermissionedDomain::visitEntry(
+    bool,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const& after)
+{
+    if (after->getType() != ltPERMISSIONED_DOMAIN)
+        return;
+    auto const& credentials = after->getFieldArray(sfAcceptedCredentials);
+    credentialsSize_ = credentials.size();
+    auto const sorted = credentials::makeSorted(credentials);
+    isUnique_ = !sorted.empty();
+
+    // If array have duplicates then all the other checks are invalid
+    isSorted_ = false;
+
+    if (isUnique_)
+    {
+        unsigned i = 0;
+        for (auto const& cred : sorted)
+        {
+            auto const& credTx = credentials[i++];
+            isSorted_ = (cred.first == credTx[sfIssuer]) &&
+                (cred.second == credTx[sfCredentialType]);
+            if (!isSorted_)
+                break;
+        }
+    }
+}
+
+bool
+ValidPermissionedDomain::finalize(
+    STTx const& tx,
+    TER const result,
+    XRPAmount const,
+    ReadView const& view,
+    beast::Journal const& j)
+{
+    if (tx.getTxnType() != ttPERMISSIONED_DOMAIN_SET || result != tesSUCCESS)
+        return true;
```

Comment:
Is this needed?

Reply:
I believe Mark put it here to be  consistent with other `finalize`  functions (they have such a check)